File: 06.bean-conditional/src/main/java/study/spring/beanconditional/v3/BooleanConditional.java
package study.spring.beanconditional.v3;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.context.annotation.Conditional;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Conditional(BooleanCondition.class)
public @interface BooleanConditional {

  boolean value();

}

File: bean-conditional/src/main/kotlin/study/spring/beanconditional/BooleanConditional.kt
package study.spring.beanconditional

import org.springframework.context.annotation.Conditional

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Conditional(BooleanCondition::class)
annotation class BooleanConditional(
    val value: Boolean,
)